server/server.js

```diff
@@ -1,35 +1,212 @@
 const express = require('express');
-const knex = require('./knex');  
-const axios = require('axios');  
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const cookieParser = require('cookie-parser');
 const cors = require('cors');
-const app = express();
-
-require('dotenv').config();
+const axios = require('axios'); 
+const knex = require('./knex'); 
+require('dotenv').config(); 
 
+const app = express();
 app.use(cors());
 app.use(express.json());
+app.use(cookieParser());
+
+
+function authenticateToken(req, res, next) {
+    const token = req.cookies.token;
+
+    if (!token) {
+        return res.status(401).json({ error: 'Access denied. No token provided.' });
+    }
+
+    jwt.verify(token, process.env.ACCESS_TOKEN_SECRET, (err, user) => {
+        if (err) {
+            return res.status(403).json({ error: 'Invalid token.' });
+        }
+        req.user = user;
+        next();
+    });
+}
```

Comment:
Nice! Great to see this in use. We will need to figure out how to send the token to the front end after we authorize the user.